Update Mockito to 2.x and JUnit to latest 4.x

TWiStErRob opened the pull request.

TWiStErRob committed:
Update JUnit and Mockito versions to latest

TWiStErRob committed:
fix deprecated imports (Matchers -> ArgumentMatchers)

TWiStErRob committed:
remove dependency on mockito-all, features are only used from mockito-core

TWiStErRob committed:
remove aliased deprecated call

TWiStErRob committed:
fix failing test

TWiStErRob committed:
replace null-allowing any() with any(Class) to prevent passing nulls

TWiStErRob commented on jfixture/src/test/java/com/flextrade/jfixture/TestFixtureBehaviours.java:
```diff
@@ -36,12 +36,14 @@ public void initialise() {
+        when(mockBehaviour.transform(mockSpecimenBuilder)).thenReturn(mockSpecimenBuilder);
         this.behaviours.add(this.mockBehaviour);
-        verify(this.mockBuilderContainer).setFinalBuilder(any(CompositeBuilder.class));
+        verify(this.mockBuilderContainer).setFinalBuilder(any(SpecimenBuilder.class));
```
based on the code this will never be `CompositeBuilder` unless the `mockSpecimenBuilder` returns one; in production this is true, but not here, since this test is mocked; probably "historical reasons"

TWiStErRob reviewed (commented):
Mockito 2.0 has a major breaking change: `any(Class)` no rejects nulls, which means that missing stubbings and/or incorrect verifications will now detect more issues.

philipwhiuk commented:
Are we happy with the breaking change here @richkeenan @cliffsun91 @rses @phutchin ?

TWiStErRob commented:
Just to clarify: there are no breaking API changes, I changed a test dependency. Mockito just got stricter and detected some false positive tests that I fixed here. I hope the new verifications are correct.

philipwhiuk commented:
Oh it's a test scope dependency change to 2.x. Fine. Merging.

philipwhiuk committed:
Merge branch 'master' into mockito_update

philipwhiuk committed:
Tidy up mockito dependency in jfixture core post merge
